tests/functional_tests/cpp_backend_tests/test_driver.py

```diff
@@ -0,0 +1,58 @@
+import os
+import subprocess
+from pathlib import Path
+
+import pytest
+
+
+def _source_dir():
+    return Path(__file__).resolve().parent
+
+
+def _build_dir(backend_dir: str):
+    return _source_dir().joinpath(f"build_{backend_dir}")
+
+
+def _execute_cmake(backend_str: str):
+    build_dir = _build_dir(backend_str)
+    os.makedirs(build_dir, exist_ok=True)
+    cmake = ["cmake", "-B", build_dir, f"-DBACKEND={backend_str}"]
+    cmake_proc = subprocess.run(cmake, cwd=_source_dir())
+    cmake_proc.check_returncode()
+    return True
```

Comment:
```suggestion
    subprocess.run(cmake, cwd=_source_dir(), check=True)
```

Reply:
Or maybe if you want to make clearer that this is the tested code:
```suggestion
    assert subprocess.run(cmake, cwd=_source_dir()).returncode == 0
```